File: identhub/src/main/java/de/solarisbank/identhub/session/data/verification/phone/VerificationPhoneNetworkDataSource.java
package de.solarisbank.identhub.session.data.verification.phone;

import de.solarisbank.identhub.session.data.TransactionAuthenticationNumber;
import de.solarisbank.identhub.session.data.verification.phone.model.VerificationPhoneResponse;
import io.reactivex.Single;

public interface VerificationPhoneNetworkDataSource {
    Single<VerificationPhoneResponse> postAuthorize();

    Single<VerificationPhoneResponse> postConfirm(final TransactionAuthenticationNumber tan);
}

File: startup/src/main/java/de/solarisbank/identhub/startup/data/VerificationPhoneNetworkDataSource.kt
package de.solarisbank.identhub.startup.data

import de.solarisbank.identhub.startup.model.TransactionAuthenticationNumber
import de.solarisbank.identhub.startup.model.VerificationPhoneResponse
import io.reactivex.Single

interface VerificationPhoneNetworkDataSource {
    fun postAuthorize(): Single<VerificationPhoneResponse>
    fun postConfirm(tan: TransactionAuthenticationNumber): Single<VerificationPhoneResponse>
}